src/types.ts

```diff
@@ -12,30 +12,140 @@ export interface PlayerType {
   playerIndex: number;
   port: number;
   characterId: number | null;
-  characterColor: number | null;
-  startStocks: number | null;
   type: number | null;
+  startStocks: number | null;
+  characterColor: number | null;
+  teamShade: number | null;
+  handicap: number | null;
   teamId: number | null;
+  staminaMode: boolean | null;
+  silentCharacter: boolean | null;
+  invisible: boolean | null;
+  lowGravity: boolean | null;
+  blackStockIcon: boolean | null;
+  metal: boolean | null;
+  startOnAngelPlatform: boolean | null;
+  rumbleEnabled: boolean | null;
+  cpuLevel: number | null;
+  offenseRatio: number | null;
+  defenseRatio: number | null;
+  modelScale: number | null;
   controllerFix: string | null;
   nametag: string | null;
   displayName: string;
   connectCode: string;
   userId: string;
 }
 
-export enum GameMode {
+export enum GameModeType {
   VS = 0x02,
+  CLASSIC = 0x03,
+  ADVENTURE = 0x04,
+  ALL_STAR = 0x05,
   ONLINE = 0x08,
+  CAMERA = 0x0a,
+  TARGET_TEST = 0x0f,
+  SUPER_SUDDEN_DEATH = 0x10,
+  INVISIBLE_MELEE = 0x11,
+  SLO_MO_MELEE = 0x12,
+  LIGHTNING_MELEE = 0x13,
+  TRAINING = 0x1c,
+  TINY_MELEE = 0x1d,
+  GIANT_MELEE = 0x1e,
+  STAMINA_MODE = 0x1f,
+  HOME_RUN_CONTEST = 0x20,
+  TEN_MAN_MELEE = 0x21,
+  HUNDRED_MAN_MELEE = 0x22,
+  THREE_MINUTE_MELEE = 0x23,
+  FIFTEEN_MINUTE_MELEE = 0x24,
+  ENDLESS_MELEE = 0x25,
+  CRUEL_MELEE = 0x26,
+  FIXED_CAMERA = 0x2a,
+  SINGLE_BUTTON = 0x2c,
 }
 
 export interface GameStartType {
   slpVersion: string | null;
+  timerType: TimerType | null;
+  gameMode: GameModeType | null;
+  friendlyFireEnabled: boolean | null;
+  bombRainEnabled: boolean | null;
   isTeams: boolean | null;
-  isPAL: boolean | null;
+  itemSpawnBehavior: ItemSpawnBehaviorType | null;
+  selfDestructScoreValue: number | null;
   stageId: number | null;
+  startingTimerSeconds: number | null;
+  enabledItems: number | null;
+  damageRatio: number | null;
   players: PlayerType[];
-  scene: number | null;
-  gameMode: GameMode | null;
+  randomSeed: number | null;
+  isPAL: boolean | null;
+  isFrozenPS: boolean | null;
+  minorScene: number | null;
+  majorScene: number | null;
+}
+
+export enum TimerType {
+  NONE = 0b00,
+  DECREASING = 0b10,
+  INCREASING = 0b11,
+}
+
+export enum ItemSpawnBehaviorType {
+  OFF = 0xff,
+  VERY_LOW = 0x00,
+  LOW = 0x01,
+  MEDIUM = 0x02,
+  HIGH = 0x03,
+  VERY_HIGH = 0x04,
+}
+
+export enum EnabledItemType {
+  METAL_BOX = 2 ** 0,
+  CLOAKING_DEVICE = 2 ** 1,
+  POKEBALL = 2 ** 2,
+  // Bits 4 through 8 of item bitfield 1 are unknown
+  UNKNOWN_ITEM_BIT_4 = 2 ** 3,
+  UNKNOWN_ITEM_BIT_5 = 2 ** 4,
+  UNKNOWN_ITEM_BIT_6 = 2 ** 5,
+  UNKNOWN_ITEM_BIT_7 = 2 ** 6,
+  UNKNOWN_ITEM_BIT_8 = 2 ** 7,
```

Comment:
How come you went with defining the values here as `2 ** X` instead of just writing out the number?

Reply:
I did it this way for readability. I don't want to write out the numbers in decimal since they get pretty large and it may not be immediately obvious to someone digging through the library where the numbers come from - `2^39` is `549755813888`. Writing out as binary works but is also long - 40 characters for the above number.